modules/mediawiki/files/bin/deploy-mediawiki.py

```diff
@@ -167,134 +236,151 @@ def _construct_git_pull(repo: str, submodules: bool = False, branch: Optional[st
+            if not tag_is_valid(language):
+                raise ValueError(f'{language} is not a valid language.')
+        lang = f' --lang={lang}'
+    else:
+        lang = ''
+    return WikiCommand(f'/srv/mediawiki/w/maintenance/rebuildLocalisationCache.php{lang} --quiet', db)
+
+
+def _get_git_commands(world: bool, pull: str | None, branch: str | None) -> list[str]:
+    stage: list[str] = []  # commands
+    pulls: list[str] = []  # repos
+    if world and not pull:
+        pulls = ['world']
+    if pull:
+        pulls = str(pull).split(',')
+    if world and 'world' not in pulls:
+        pulls.append('world')
+    if pulls:
+        for repo in pulls:
+            if repo == 'world':
+                sm = True
+            else:
+                sm = False
+            try:
+                stage.append(_construct_git_pull(repo, submodules=sm, branch=branch))
+            except KeyError:
+                print(f'Failed to pull {repo} due to invalid name')
+    return stage
+
+
+def prep(args: argparse.Namespace) -> deploymap:
+    deploymentmap: deploymap = {
+        'servers': [],
+        'doworld': args.world,
+        'loginfo': {},
+        'branch': '',
+        'nolog': args.nolog,
+        'force': args.force,
+        'port': args.port,
+        'ignoretime': args.ignoretime,
+        'debugurl': '',
+        'commands': {
+            'stage': [],
+            'rsync': [],
+            'postinstall': [],
+            'rebuild': [],
+        },
+        'remote': {
+            'paths': [],
+            'files': [],
+        },
+    }
     envinfo = get_environment_info()
-    servers = get_server_list(envinfo, args.servers)
+    deploymentmap['debugurl'] = envinfo['wikiurl']
+    deploymentmap['servers'] = get_server_list(envinfo, args.servers)
     options = {'config': args.config, 'world': args.world, 'landing': args.landing, 'errorpages': args.errorpages}
-    exitcodes = []
     loginfo = {}
-    rsyncpaths = []
-    rsyncfiles = []
-    rsync = []
-    rebuild = []
-    postinstall = []
-    stage = []
     for arg in vars(args).items():
         if arg[1] is not None and arg[1] is not False:
             loginfo[arg[0]] = arg[1]
-    synced = loginfo['servers']
-    if HOSTNAME in servers:
-        del loginfo['servers']
-        text = f'starting deploy of "{str(loginfo)}" to {synced}'
-        if not args.nolog:
+    deploymentmap['loginfo'] = loginfo
+    deploymentmap['commands']['stage'] = _get_git_commands(args.world, args.pull, args.branch)
+    for option in options:  # configure rsync & custom data for repos
+        if options[option]:
+            if option == 'world':  # install steps for w
+                deploymentmap['commands']['rebuild'].append(WikiCommand('MW_INSTALL_PATH=/srv/mediawiki-staging/w php /srv/mediawiki-staging/w/extensions/MirahezeMagic/maintenance/rebuildVersionCache.php --save-gitinfo --conf=/srv/mediawiki-staging/config/LocalSettings.php', envinfo['wikidbname']))
```

Comment:
Would this not essentually be `sudo -u www-data php MW_INSTALL_PATH=... php...`?

Reply:
Yeah, I'll fix that